src/packets.rs

```diff
@@ -1050,35 +1050,54 @@ pub struct HandshakeResponse {
 }
 
 impl HandshakeResponse {
+    /// Builds a payload following [Protocol::HandshakeResponse41](https://dev.mysql.com/doc/internals/en/connection-phase-packets.html#packet-Protocol::HandshakeResponse) specification
+    /// If db_name is Some but clients flags are missing CLIENT_CONNECT_WITH_DB,
+    /// the flag will be appended automatically
     pub fn new(
         scramble_buf: &Option<impl AsRef<[u8]>>,
         server_version: (u16, u16, u16),
         user: Option<&str>,
         db_name: Option<&str>,
         auth_plugin: &AuthPlugin<'_>,
-        client_flags: CapabilityFlags,
+        mut client_flags: CapabilityFlags,
         connect_attributes: &HashMap<String, String>,
     ) -> HandshakeResponse {
         let scramble = scramble_buf.as_ref().map(|x| x.as_ref()).unwrap_or(&[]);
-        let database = db_name.unwrap_or("");
 
         let collation = if server_version >= (5, 5, 3) {
             UTF8MB4_GENERAL_CI
         } else {
             UTF8_GENERAL_CI
         };
 
+        if let Some(_) = db_name {
+            client_flags = client_flags | CapabilityFlags::CLIENT_CONNECT_WITH_DB;
+        }
+
         let mut data = Vec::with_capacity(1024);
         data.write_u32::<LE>(client_flags.bits()).unwrap();
-        data.resize(data.len() + 4, 0);
+        data.extend_from_slice(&[0x00, 0x00, 0x00, 0x01]);
```

Comment:
I've changed this line because `rust_mysql_common` was sending `max-packet size` as 0.

Here is what the official MySQL client sends in `max-packet size`:

<img width="449" alt="Screen Shot 2020-05-09 at 09 19 49" src="https://user-images.githubusercontent.com/522155/81468842-e12d6200-91d9-11ea-9685-7383e6608fd0.png">

I figured we want to follow this to avoid surprises in package length.